### ISSUE TITLE
Cache go dependencies in GitHub Actions e2e tests

### ISSUE THREAD

**Issue by gjermundgaraba** [cosmos/ibc-go#6534]
The setup-go action we already use has a caching mechanism built in that we can leverage to avoid the ~2 minutes of downloading of dependencies that we do for every e2e build.

The action by default uses caching for the root-level dependencies, but needs to be told if there are go modules in sub-folder that it should be caching instead (it accepts a list). The caching mechanism is using a hash of the go.sum file as part of its key, so as long as the go.sum file is the same, we will get a cache-hit.

In other words, we can get a nice speed-up by simple adding `cache-dependency-path: 'e2e/go.sum'` as an argument to `actions/setup-go@v5`.

Documentation: https://github.com/actions/setup-go?tab=readme-ov-file#caching-dependency-files-and-build-outputs
